docs/2gp_testing.rst

```diff
@@ -0,0 +1,123 @@
+Testing with Second-Generation Packaging
+========================================
+
+CumulusCI and the CumulusCI Flow make it easy to harness the power of second-generation
+managed packages to implement an advanced, comprehensive testing process for *both* 
+first- and second-generation managed package products.
+
+This process yields multiple benefits:
+
+* You can test managed packages before merging code.
```

Comment:
I mean, we did that before, as unmanaged deployments. The key difference is that this way it's actually a managed package with the differences that go along with that (separate namespace, install script, ...). Should we spell that out?